dev/NumberBox/NumberBox.h

```diff
@@ -0,0 +1,103 @@
+    void OnIsZeroSignedPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnSignificantDigitsPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnRoundingAlgorithmPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnNumberRounderPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnSignificantDigitPrecisionPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnIncrementPrecisionPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnPlaceholderTextPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnValuePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnTextPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+
+    void OnAcceptsCalculationsPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnBasicValidationModePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnHyperScrollEnabledPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnMaxValuePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnMinMaxModePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnMinValuePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    void OnStepFrequencyPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args);
+    bool m_hasError{ false };
+
+
+private:
+
+    void OnTextBoxLostFocus(winrt::IInspectable const& sender, winrt::RoutedEventArgs const& args);
+    void ValidateInput();
+    void UpdateTextToValue();
+    void SetErrorState(ValidationState state);
+    void SetSpinButtonVisualState();
+    void OnSpinDownClick(winrt::IInspectable const& sender, winrt::RoutedEventArgs const& args);
+    void OnSpinUpClick(winrt::IInspectable const& sender, winrt::RoutedEventArgs const& args);
+    void OnNumberBoxKeyUp(winrt::IInspectable const& sender, winrt::KeyRoutedEventArgs const& args);
+    int ComputePrecisionRounderSigDigits(double newVal);
+    void OnScroll(winrt::IInspectable const& sender, winrt::PointerRoutedEventArgs const& args);
+    void StepValue(bool sign);
+    void UpdateFormatter();
+    void UpdateRounder();
+    void SetHeader();
+    void SetPlaceHolderText();
+    void EvaluateInput();
+    bool IsFormulaic(const winrt::hstring& in);
+    void NormalizeShorthandOperations();
+    BoundState GetBoundState(double val);
+    winrt::DecimalFormatter m_formatter{};
+    winrt::IncrementNumberRounder m_iRounder{};
+    winrt::SignificantDigitsNumberRounder m_sRounder{};
+    winrt::DecimalFormatter m_stepPrecisionFormatter{};
+    winrt::SignificantDigitsNumberRounder m_stepPrecisionRounder{};
+    winrt::TextBox m_TextBox{};
```

Comment:
I think @StephenLPeters may have already mentioned this, but fields that reference DependencyObject-typed things (like controls and other template parts) need to be declared as `tracker_ref<winrt::TextBox> m_textBox{this}`, etc. The tracker_ref doodad is how we make the .NET garbage collector aware of our objects and their references so that orphaned cycles can be collected.